impeller/tessellator/tessellator.cc

```diff
@@ -32,7 +32,19 @@ static TESSalloc alloc = {
 };
 
 Tessellator::Tessellator()
-    : c_tessellator_(::tessNewTess(&alloc), &DestroyTessellator) {}
+    : c_tess_alloc_(nullptr, &DestroyTessAlloc),
+      c_tessellator_(nullptr, &DestroyTessellator) {
+  {
+    CTessAlloc tess_alloc(new TESSalloc(), &DestroyTessAlloc);
+    c_tess_alloc_ = std::move(tess_alloc);
+  }
+  memcpy(c_tess_alloc_.get(), &kAlloc, sizeof(TESSalloc));
+  {
+    CTessellator tessellator(::tessNewTess(c_tess_alloc_.get()),
```

Comment:
`tessNewTess` copies the `TESSalloc` contents into the newly created tessellator and does not mutate the `alloc` argument:
https://flutter.googlesource.com/third_party/libtess2/+/refs/heads/master/Source/tess.h#84
https://flutter.googlesource.com/third_party/libtess2/+/refs/heads/master/Source/tess.c#610

If you want to clarify the lifetime of this structure, then it should be sufficient to declare the `TESSalloc` as a local variable here.

Reply:
Done.  The code as it was written is conforming to the api and docstrings since it takes a mutable pointer and has no mention of what it does to it.  libTess2 will never be updated so it should be safe to pass a pointer to something on the stack.